openmc/opencg_compatible.py

```diff
@@ -881,16 +881,28 @@ def get_opencg_lattice(openmc_lattice):
     universes = openmc_lattice.universes
     outer = openmc_lattice.outer
 
+    # Convert 2D dimension to 3D for OpenCG
+    if len(dimension) == 2:
+        new_dimension = np.ones(3, dtype=np.int)
+        new_dimension[:2] = dimension
+        dimension = new_dimension
+
+    # Convert 2D pitch to 3D for OpenCG
     if len(pitch) == 2:
-        new_pitch = np.ones(3, dtype=np.float64) * np.inf
+        new_pitch = np.ones(3, dtype=np.float64) * np.finfo(np.float64).max
         new_pitch[:2] = pitch
         pitch = new_pitch
 
+    # Convert 2D lower left to 3D for OpenCG
     if len(lower_left) == 2:
-        new_lower_left = np.ones(3, dtype=np.float64)
+        new_lower_left = np.ones(3, dtype=np.float64) * np.finfo(np.float64).min
         new_lower_left[:2] = lower_left
         lower_left = new_lower_left
 
+    # Convert 2D universes array to 3D for OpenCG
+    if len(universes.shape) == 2:
```

Comment:
In some input files, the universes attribute is set as a list of Universe objects (e.g. build-xml.py for the simple lattice example input file). The lattice universes setter only checks the type of object in the list/ndarray, but does not convert lists to ndarrays. If `universes` is a list, this will error out. The solution I found to this is to just change line 810 of `universes.py` to:

    self._universes = np.asarray(universes)